src/lib/source-handlers/azure/list-repos.ts

```diff
@@ -47,22 +48,16 @@ async function getRepos(
   token: string,
 ): Promise<AzureRepoData[]> {
   const repoList: AzureRepoData[] = [];
-  const data = await limiter.schedule(() =>
-    needle(
-      'get',
-      `${url}/${orgName}/` +
-        encodeURIComponent(project) +
-        '/_apis/git/repositories?api-version=4.1',
-      {
-        headers: { Authorization: `Basic ${base64.encode(':' + token)}` },
-      },
-    ),
+  const headers: OutgoingHttpHeaders = { Authorization: `Basic ${base64.encode(':' + token)}` }
+  const data = await requestWithRateLimitRetries(
+    'get',
+    `${url}/${orgName}/` +
+      encodeURIComponent(project) +
+      '/_apis/git/repositories?api-version=4.1',
+      headers,
+    limiter
   );
-  if (data.statusCode != 200) {
```

Comment:
you need to keep this, only the `const = data` line should change. Even after 7 retries it can be a non 200 status code so we need to handle it

Reply:
Done